.github/workflows/unit-tests.yml

```diff
@@ -76,24 +78,21 @@ jobs:
           mongod &
 
       - name: Setup Python
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: checkout repo
-        uses: actions/checkout@v3
-
       - name: install requirements
         run: |
-          sudo make test-requirements
+          sudo env "PATH=$PATH" make test-requirements
           if [[ "${{ matrix.django-version }}" != "pinned" ]]; then
-            sudo pip install "django~=${{ matrix.django-version }}.0"
-            sudo pip check # fail if this test-reqs/Django combination is broken
+            sudo env "PATH=$PATH" pip install "django~=${{ matrix.django-version }}.0"
+            sudo env "PATH=$PATH" pip check # fail if this test-reqs/Django combination is broken
           fi
 
       - name: list installed package versions
         run: |
-          sudo pip freeze
+          sudo env "PATH=$PATH" pip freeze
```

Comment:
> /opt/hostedtoolcache folder contains the previously installed Python and it is not presents in the $PATH with sudo. Therefore, when you run a Python command with sudo, you cannot get Python installed by setup-python.

https://github.com/actions/setup-python/issues/225#issuecomment-982734703

Reply:
Can we resolve it without `sudo`